pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -2052,6 +2052,69 @@ def usage(self) -> _usage.Usage:
         """Get usage statistics for the run so far, including token usage, model requests, and so on."""
         return self._graph_run.state.usage
 
+    async def validate_structured_output(
```

Comment:
This is basically a copy-paste of the other method in `_agent_graph`, I suggest moving it to a shared util.